Keen.NetStandard/Dataset/Datasets.cs

```diff
@@ -0,0 +1,306 @@
+            new JsonSerializerSettings
+            {
+                ContractResolver = new DefaultContractResolver
+                {
+                    NamingStrategy = new SnakeCaseNamingStrategy()
+                },
+                DefaultValueHandling = DefaultValueHandling.IgnoreAndPopulate,
+                Formatting = Formatting.None
+            };
+
+        private readonly IKeenHttpClient _keenHttpClient;
+        private readonly string _cachedDatasetRelativeUrl;
+        private readonly string _masterKey;
+
+
+        internal Datasets(IProjectSettings prjSettings,
+                          IKeenHttpClientProvider keenHttpClientProvider)
+        {
+            if (null == prjSettings)
+            {
+                throw new ArgumentNullException(nameof(prjSettings),
+                                                "Project Settings must be provided.");
+            }
+
+            if (null == keenHttpClientProvider)
+            {
+                throw new ArgumentNullException(nameof(keenHttpClientProvider),
+                                                "A KeenHttpClient provider must be provided.");
+            }
+
+            if (string.IsNullOrWhiteSpace(prjSettings.KeenUrl) ||
+                !Uri.IsWellFormedUriString(prjSettings.KeenUrl, UriKind.Absolute))
+            {
+                throw new KeenException(
+                    "A properly formatted KeenUrl must be provided via Project Settings.");
+            }
+
+            var serverBaseUrl = new Uri(prjSettings.KeenUrl);
+            _keenHttpClient = keenHttpClientProvider.GetForUrl(serverBaseUrl);
+            _cachedDatasetRelativeUrl =
+                KeenHttpClient.GetRelativeUrl(prjSettings.ProjectId,
+                                              KeenConstants.DatasetsResource);
+
+            _masterKey = prjSettings.MasterKey;
+        }
+
+        public async Task<JObject> GetResultsAsync(string datasetName,
+                                                   string indexBy,
+                                                   string timeframe)
+        {
+            if (string.IsNullOrWhiteSpace(datasetName))
+            {
+                throw new KeenException("A dataset name is required.");
+            }
+
+            if (string.IsNullOrWhiteSpace(indexBy))
+            {
+                throw new KeenException("A value to index by is required.");
+            }
+
+            if (string.IsNullOrWhiteSpace(timeframe))
+            {
+                throw new KeenException("A timeframe by is required.");
+            }
+
+            if (string.IsNullOrWhiteSpace(_masterKey))
+            {
+                throw new KeenException("An API masterkey is required to get dataset results.");
+            }
+
+            var datasetResultsUrl = $"{GetDatasetUrl(datasetName)}/results";
+
+            var url = $"{datasetResultsUrl}?index_by={indexBy}&timeframe={timeframe}";
```

Comment:
Yeah here and elsewhere we need to verify this before merging.

Reply:
`index_by` can be any JSOn member name, I think, which in theory can be all sorts of things that need encoding.
`timeframe` when retrieving results can be an absolute timeframe (not just relative) I think, so it could have things like a colon, so also needs encoding.
`limit` shouldn't need encoding since it's just an int.